linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -166,18 +166,39 @@ final class Emitter private (config: CommonPhaseConfig,
       )
 
       for ((className, requiredMethodNames) <- requiredDefaultMethodDecls) {
-        val memberMethods = unit.classDefs
+        val methods = unit.classDefs
           .find(_.encodedName == className)
-          .fold[List[Versioned[MethodDef]]](Nil)(_.memberMethods)
+          .fold[List[Versioned[MethodDef]]](Nil)(_.methods)
         for {
           methodName <- requiredMethodNames
-          if !memberMethods.exists(_.value.name.encodedName == methodName)
+          if !methods.exists { m =>
+            m.value.flags.namespace == MemberNamespace.Public &&
+            m.value.name.encodedName == methodName
+          }
         } {
           implicit val pos = NoPosition
           val field = jsGen.envField("f", className + "__" + methodName).ident
           builder.addJSTree(js.VarDef(field, None))
         }
       }
+
+      // CloneNotSupportedException's nullary constructor
+      locally {
+        val className = "jl_CloneNotSupportedException"
+        val ctorName = "init___"
+        val ctorIsDefined = unit.classDefs
+          .find(_.encodedName == className)
+          .fold(false)(_.methods.exists { m =>
+            System.err.println(m.value)
+            m.value.flags.namespace == MemberNamespace.Constructor &&
+            m.value.encodedName == ctorName
+          })
+        if (!ctorIsDefined) {
+          implicit val pos = NoPosition
+          val field = jsGen.envField("ct", className + "__" + ctorName).ident
+          builder.addJSTree(js.VarDef(field, None))
+        }
+      }
```

Comment:
Hum, not sure I understand what is going on here... Why do we need this now?

Reply:
It is necessary now because constructors became top-level functions, so they are statically referenced, and therefore can cause GCC compile errors if they are called but not defined. Previously, constructors were instance methods, so they were not statically referenced and would not cause GCC errors.